### PR TITLE
Changed WASD to keybindings

### LINKED ISSUE #3633: Ability to keybind movement keys to other keys.
As a player with a non qwerty keyboard, playing bee is a downright nightmare of having to switch between a french layout and english layout mid game. It leads to a lot of confusion, annoyance and anger, for no real reason as a keybind menu is already coded in - but for some reason, I cannot change the keybinds of my movement keys.

I've been playing bee for a month, and despite wanting really strongly to stay, this one missing feature is legitimately ruining my experience to a point where i just want to stop playing sometimes.

For the love of all that is holy, please. Let us keybind all possible keys.

### PR DESCRIPTION
**Pull request by stinky-lizard**
## About The Pull Request

Removes hardcoded WASD keys and replaces them with WASD keybinds. Now, instead of just using WASD, you can use any key on the keyboard - they'll even fire alongside other keybinds. Uses the existing keybind system. WASD is still default.

## Why It's Good For The Game

Allows people with non-QWERTY keyboard layouts to play much more comfortably. In turn, fixes #3633 

## Changelog
:cl:
add: Made WASD movement keys rebindable
/:cl:

### CODE REVIEW

**Comment by Kapu1178** [comment]
Does this maintain non-hotkey mode movement?

**Comment by stinky-lizard** [comment]
By the way, did a quick test on a local server, and worked smoothly. Might not hold up to lag though? Rebinding WASD to -P[] worked fine.

**Comment by stinky-lizard** [comment]
> 
> 
> Does this maintain non-hotkey mode movement?

If you mean the arrow keys, yes, it does.

**Comment by stinky-lizard** [comment]
Here's a quick video showcasing it: https://youtu.be/8NYEZZGI8NA

**Comment by francinum** [comment]
Please use the *exact phrasing* of `closes #000` or `fixes #000`, with each individual issue having it's own `closes/fixes/resolves`
Github doesn't track closure properly if you don't do this.

**Comment by stinky-lizard** [comment]
Oh, sorry. Edited.

**Comment by francinum** [comment]
Looks good overall.

File: code/controllers/subsystem/input.dm [hunk 1]
```diff
@@ -95,7 +95,7 @@ SUBSYSTEM_DEF(input)
 // For initially setting up or resetting to default the movement keys
 /datum/controller/subsystem/input/proc/setup_default_movement_keys()
 	var/static/list/default_movement_keys = list(
-		"W" = NORTH, "A" = WEST, "S" = SOUTH, "D" = EAST,				// WASD
+//		"W" = NORTH, "A" = WEST, "S" = SOUTH, "D" = EAST,				// WASD are now keybinds, not hard-coded.
```
Comment (KubeRoot) [inline code/controllers/subsystem/input.dm]: The only thing I'm unsure about is if this should stay commented out, or just be removed.

**Review by KubeRoot — DISMISSED** [review]
Looks good, tested with both hotkey and old-style input.

File: code/controllers/subsystem/input.dm (hunk 1 above)
Comment (stinky-lizard) [reply code/controllers/subsystem/input.dm]: That's fair. I've removed it.

**Review by KubeRoot — CHANGES_REQUESTED** [review]
This PR needs to be updated so the new keybindings match how keybindings are coded now: https://github.com/BeeStation/BeeStation-Hornet/pull/5672

This generally means each keybinding no longer does its own checks `if(!user.mob) return`, but calls parent and checks the return value of that.

I also didn't notice earlier, but it seems keybindings should return `TRUE` if the input was handled correctly, which none of the new keybindings do; is there a specific reason for that?

**Comment by stinky-lizard** [comment]
> This PR needs to be updated so the new keybindings match how keybindings are coded now: #5672
> 
> This generally means each keybinding no longer does its own checks `if(!user.mob) return`, but calls parent and checks the return value of that.
> 
> I also didn't notice earlier, but it seems keybindings should return `TRUE` if the input was handled correctly, which none of the new keybindings do; is there a specific reason for that?

I've updated the new keybinds to match how the others act in mob.dm - neither of the other two changed files in this commit  (input.dm & preferences_savefile.dm) were touched by #5672, so I figured everything in them would be communicated with in the same way, so I think they should be fine without changes?

Also, all the keybindings still have `if(!user.mob) return`... I checked and that's not just from my commits, they're there in the main fork. So I kept those in my keybinds as well.

There were some other oddities as well, like how prevent_movement didn't return `TRUE` at all, or how `prevent_movement/up` checks its parent but `toggle_move_intent/up` doesn't. Is that intentional? I barely have a grasp on how the whole system works, I'm mostly copying the other keybinds, but y'know, can you blame me?

Either way, tested in a local server and it worked fine. ¯\_(ツ)_/¯

**Comment by stinky-lizard** [comment]
Oh, also, I noticed this fixes a bit of a movement bug: without this PR, moving with WASD and switching to non-hotkey mode while moving keeps you moving in that direction until you return to hotkey mode and press the button again - pressing the arrow keys wont fix it. With it, switching from hotkey mode still keeps you moving, but now you only have to press the corresponding direction on the arrow keys.

**Comment by KubeRoot** [comment]
> Also, all the keybindings still have `if(!user.mob) return`... I checked and that's not just from my commits, they're there in the main fork. So I kept those in my keybinds as well.

Uh oh, certainly not all the keybindings, and I believe that's my fault, gonna have to look into it. If I remember correctly, these should only be there for very peculiar keybindings that require checks that don't fit in their type. The reason for this, `keybinding/can_use` is ran to decide if `/down` or `/up` is actually executed, and those are overriden on subtypes like `keybinding/mob` to include the checks previously duplicated in each `down`/`up`.

**Comment by stinky-lizard** [comment]
> Uh oh, certainly not all the keybindings, and I believe that's my fault, gonna have to look into it. If I remember correctly, these should only be there for very peculiar keybindings that require checks that don't fit in their type.

Upon closer inspection, not all of them have it, but most do, at least in `mob.dm`.

File: code/datums/keybinding/mob.dm [hunk 2]
```diff
@@ -1,8 +1,104 @@
...
+	name = "move_north"
+	full_name = "Move North"
+	description = ""
+	keybind_signal = COMSIG_KB_MOB_MOVENORTH_DOWN
+
+/datum/keybinding/mob/move_north/down(client/user)
+	. = ..()
+	if(.)
+		return
+	if(!user.mob) return
```
Comment (utilbird) [inline code/datums/keybinding/mob.dm:17]: Properly newline indent the returns for consistency with the rest of the codebase

Comment (stinky-lizard) [reply code/datums/keybinding/mob.dm:17]: On line 17? The rest of the file has the same indentation as 14-17, should I change those as well?

**Review by PowerfulBacon — APPROVED** [review]
Works fine

**Pull request merged**